board/safety/safety_tesla.h

```diff
@@ -0,0 +1,350 @@
+// board enforces
+//   in-state
+//      accel set/resume
+//   out-state
+//      cancel button
+//      regen paddle
+//      accel rising edge
+//      brake rising edge
+//      brake > 0mph
+
+// 2m/s are added to be less restrictive
+const struct lookup_t TESLA_LOOKUP_ANGLE_RATE_UP = {
+    {2., 7., 17.},
+    {5., .8, .25}};
+
+const struct lookup_t TESLA_LOOKUP_ANGLE_RATE_DOWN = {
+    {2., 7., 17.},
+    {5., 3.5, .8}};
+
+const struct lookup_t TESLA_LOOKUP_MAX_ANGLE = {
+    {2., 29., 38.},
+    {410., 92., 36.}};
+
+const int TESLA_RT_INTERVAL = 250000; // 250ms between real time checks
+
+struct sample_t tesla_angle_meas; // last 3 steer angles
+
+// state of angle limits
+int tesla_desired_angle_last = 0; // last desired steer angle
+int16_t tesla_rt_angle_last = 0.; // last real time angle
+uint32_t tesla_ts_angle_last = 0;
+
+int tesla_controls_allowed_last = 0;
+int steer_allowed = 1;
+
+int tesla_brake_prev = 0;
+int tesla_gas_prev = 0;
+int tesla_speed = 0;
+int current_car_time = -1;
+int time_at_last_stalk_pull = -1;
+int eac_status = 0;
+
+int tesla_ignition_started = 0;
+
+// interp function that holds extreme values
+float tesla_interpolate(struct lookup_t xy, float x)
```

Comment:
just moved a generic interpolation function to safety.h I recommend using that one.